jlm/rvsdg/bitstring/type.cpp

```diff
@@ -28,16 +28,41 @@ bittype::operator==(const jlm::rvsdg::type & other) const noexcept
   return type != nullptr && this->nbits() == type->nbits();
 }
 
-std::unique_ptr<jlm::rvsdg::type>
+std::shared_ptr<const jlm::rvsdg::type>
 bittype::copy() const
 {
-  return std::unique_ptr<jlm::rvsdg::type>(new bittype(*this));
+  return std::make_shared<bittype>(nbits_);
 }
 
-const bittype bit1(1);
-const bittype bit8(8);
-const bittype bit16(16);
-const bittype bit32(32);
-const bittype bit64(64);
+std::shared_ptr<const bittype>
+bittype::Create(std::size_t nbits)
+{
+  static const bittype static_instances[65] = {
+    bittype(0),  bittype(1),  bittype(2),  bittype(3),  bittype(4),  bittype(5),  bittype(6),
```

Comment:
Could we leave a comment for `bittype(0)` here similarly to `ctltype(0)`?

Reply:
Well, bitstring(0) is not technically illegal, it would just be an empty type that does not store any value (like "std::nullopt_t") -- I am more inclined of removing the throw below.